utils/kubernetes/healthcheck.go

```diff
@@ -0,0 +1,104 @@
+package kubernetes
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/version"
+)
+
+var minKubernetesVersion = [3]int{1, 12, 0}
+var minKubectlVersion = minKubernetesVersion
+
+//GetK8sVersionInfo returns version.Info for the Kubernetes cluster.
+func GetK8sVersionInfo(client *Client) (*version.Info, error) {
+	return client.KubeClient.Discovery().ServerVersion()
+}
+
+func CheckK8sVersion(versionInfo *version.Info) error {
+	apiVersion, err := getK8sVersion(versionInfo.String())
+	if err != nil {
+		return err
+	}
+
+	if !isCompatibleVersion(minKubernetesVersion, apiVersion) {
+		return fmt.Errorf("kubernetes is on version [%d.%d.%d], but version [%d.%d.%d] or more recent is required",
+			apiVersion[0], apiVersion[1], apiVersion[2],
+			minKubernetesVersion[0], minKubernetesVersion[1], minKubernetesVersion[2])
+	}
+
+	return nil
+}
+
+func getK8sVersion(versionString string) ([3]int, error) {
+	var version [3]int
+	var revisionSeparator = regexp.MustCompile("[^0-9.]")
+
+	justTheVersionString := strings.TrimPrefix(versionString, "v")
+	justTheMajorMinorRevisionNumbers := revisionSeparator.Split(justTheVersionString, -1)[0]
+	split := strings.Split(justTheMajorMinorRevisionNumbers, ".")
+
+	if len(split) < 3 {
+		return version, fmt.Errorf("unknown version string format [%s]", versionString)
+	}
+
+	for i, segment := range split {
+		v, err := strconv.Atoi(strings.TrimSpace(segment))
+		if err != nil {
+			return version, fmt.Errorf("unknown version string format [%s]", versionString)
+		}
+		version[i] = v
+	}
+
+	return version, nil
+}
+
+// CheckKubectlVersion validates whether the installed kubectl version is
+// running a minimum kubectl version.
+func CheckKubectlVersion() error {
+	cmd := exec.Command("kubectl", "version", "--client", "--short")
+	bytes, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+
+	clientVersion := fmt.Sprintf("%s\n", bytes)
+	kubectlVersion, err := parseKubectlShortVersion(clientVersion)
+	if err != nil {
+		return err
+	}
+
+	if !isCompatibleVersion(minKubectlVersion, kubectlVersion) {
+		return fmt.Errorf("kubectl is on version [%d.%d.%d], but version [%d.%d.%d] or more recent is required",
+			kubectlVersion[0], kubectlVersion[1], kubectlVersion[2],
+			minKubectlVersion[0], minKubectlVersion[1], minKubectlVersion[2])
+	}
+
+	return nil
+}
+
+func isCompatibleVersion(minimalRequirementVersion [3]int, actualVersion [3]int) bool {
+	if minimalRequirementVersion[0] < actualVersion[0] {
+		return true
+	}
+
+	if (minimalRequirementVersion[0] == actualVersion[0]) && minimalRequirementVersion[1] < actualVersion[1] {
+		return true
+	}
+
+	if (minimalRequirementVersion[0] == actualVersion[0]) && (minimalRequirementVersion[1] == actualVersion[1]) && (minimalRequirementVersion[2] <= actualVersion[2]) {
+		return true
+	}
+
+	return false
+}
+
+var semVer = regexp.MustCompile("(v[0-9]+.[0-9]+.[0-9]+)")
```

Comment:
@piyushsingariya, Any reason for having this outside the parseKubectlShortVersion() ? Is there a plan to use it in any other function? If not, we can move it inside the function.

Reply:
yeah sure, good idea.